### ISSUE TITLE
Enable irilyth's fanatic

### ISSUE THREAD

**Issue by cgolubi1** [buttonmen-dev/buttonmen#2882]
As discussed on #2776, we have a hypothesis that we can enable irilyth's fanatic for use on the prod site, without changes to the hit table logic, by setting a limit of `memory_limit=32M` on all site PHP configurations, including production.  The idea is that:
* 32M should be enough to handle all 3-round irilyth games against defined buttons, including ones involving aux dice
* For cases the above doesn't handle, if a game with irilyth's fanatic is expensive enough that it would destabilize the site, the memory limit might cause that game to fail to load, but will avoid distruptions to or crashes of the site

Implement this idea, and test it on a loaded dev site (meaning, one against which a bot is playing a bunch of irilyth games in the background) to validate the assertion that site experience will be reasonable for players.

**Comment by PoshFrosh** [comment]
Would it be okay if I enter and play irilyth's button as a custom recipe on staging?

**Comment by cgolubi1** [comment]
> Would it be okay if I enter and play irilyth's button as a custom recipe on staging?

It's okay with me.

I think it's worth flagging that resolving #2928 is my most critical priority currently (we have a hard deadline of mid-August by which it *must* be done or the site will become unavailable, and also not having done it is costing me money), so anything else anyone is working on using dev or staging sites, i will shunt to the side if needed to unblock that (including e.g. marking games BROKEN, changing recipes, whatever).  But as long as you don't mind that, have at it.

**Comment by PoshFrosh** [comment]
Ah, I had no idea. Looks like it got taken care of, so that's good.

I guess what I was really asking is:
I read about the idea of safeguards (like setting a 32M limit on hit table searches? Sorry if I'm stating that wrong) that would contain irilyth issues to breaking the current game rather than freezing the site.
Are any of these "safeguards" currently in place on staging? Or could I potentially freeze up staging by playing irilyth's button there?
Sorry, I was temporarily not active here when the irilyth button caused issues and I'm just trying to piece things together from reading historical comments etc.

**Comment by cgolubi1** [comment]
No, we didn't set those limits yet.  [issue=2881] is the issue where we discussed doing so.

If we enable irilyth's fanatic on staging, players can make things so that if a particular game is loaded, it freezes the site for everyone for awhile (and then it will come back).

All of this kind of fell off the radar because deployments broke 16 months ago, followed by the MySQL version emergency, so we pretty much haven't been able to do any development players care about since February 2023.  Now we're unblocked from doing so again, so i guess my ask would be: what is your number one priority in terms of site development?  Maybe we should start there, and go down the list.

**Comment by PoshFrosh** [comment]
@cgolubi1 I'm just one person, but if I'm reading this right it sounds like you are asking my opinion.

My current priority is testing my future fanatic recipe, so what would be most useful to me is getting the responder test built from [this post](https://github.com/buttonmen-dev/buttonmen/issues/2240#issuecomment-2218050778) so that #2240  can move along and ultimately that I can test %D() in staging prior to finalizing my recipe.

That being said, I also have some priorities which are less selfish: I would like to see @irilyth 's button enabled (this issue #2882) and @blackshadowshade 's as well (#2772 ), since it's seems like we are getting close on those and I think having more people's fanatics playable on this site fosters community.

**Comment by cgolubi1** [comment]
Yep, i was asking for your opinion!  Knowing that your top priority for your fanatic is unblocking Doppelganger + Radioactive is helpful.  I expect that's a combination that will take a few tries to get working, but it's certainly doable.  Thanks!

**Comment by PoshFrosh** [comment]
@cgolubi1  I don't like asking for stuff for myself, so thanks for asking for my opinion directly. I really appreciate your time and consideration. Thanks.

As for the "go down the list" part, the other two possibilities I'm investigating are:
Zero option dice which is part of aforementioned #2772 and
Dm() which is #2637 

I'm relatively certain about %D (which is why it's my top pick), but these other two I'm less certain about without being able to test them, which I currently cannot. An advantage to 2772 is that it will also get us closer to enabling the existing BSS fanatic.

Thanks again, and I'm just glad to see proper development spin back up on the site no matter what your next priority is. 😀
